Auth0/CredentialsManager.swift

```diff
@@ -158,40 +157,60 @@ public struct CredentialsManager {
+            guard self.hasExpired(credentials) ||
+                    self.willExpire(credentials, within: minTTL) ||
+                    self.hasScopeChanged(credentials, from: scope) else {
+                        self.dispatchGroup.leave()
+                        return callback(nil, credentials)
+                    }
+            guard let refreshToken = credentials.refreshToken else {
+                self.dispatchGroup.leave()
+                return callback(.noRefreshToken, nil)
+            }
+            self.authentication
+                .renew(withRefreshToken: refreshToken, scope: scope)
+                .parameters(parameters)
+                .start { result in
+                    switch result {
+                    case .success(let credentials):
+                        let newCredentials = Credentials(accessToken: credentials.accessToken,
+                                                         tokenType: credentials.tokenType,
+                                                         idToken: credentials.idToken,
+                                                         refreshToken: credentials.refreshToken ?? refreshToken,
+                                                         expiresIn: credentials.expiresIn,
+                                                         scope: credentials.scope)
+                        if self.willExpire(newCredentials, within: minTTL) {
+                            let accessTokenLifetime = Int(credentials.expiresIn.timeIntervalSinceNow)
+                            // TODO: On the next major add a new case to CredentialsManagerError
+                            let error = NSError(domain: "The lifetime of the renewed Access Token (\(accessTokenLifetime)s) is less than minTTL requested (\(minTTL)s). Increase the 'Token Expiration' setting of your Auth0 API in the dashboard or request a lower minTTL",
+                                                code: -99999,
+                                                userInfo: nil)
+                            self.dispatchGroup.leave()
+                            callback(.failedRefresh(error), nil)
+                        } else {
+                            _ = self.store(credentials: newCredentials)
+                            self.dispatchGroup.leave()
+                            callback(nil, newCredentials)
+                        }
+                    case .failure(let error):
+                        self.dispatchGroup.leave()
                         callback(.failedRefresh(error), nil)
-                    } else {
-                        _ = self.store(credentials: newCredentials)
-                        callback(nil, newCredentials)
                     }
-                case .failure(let error):
-                    callback(.failedRefresh(error), nil)
                 }
-            }
+        }
+
+        if DispatchQueue.getSpecific(key: queueKey) != queueLabel {
+            self.dispatchGroup.wait()
```

Comment:
This blocks until `dispatchGroup.leave()` is called, and thus makes the whole operation run serially.